src/core/ready.js

```diff
@@ -49,11 +47,28 @@ jQuery.extend( {
 			return;
 		}
 
-		// If there are functions bound, to execute
-		readyList.resolveWith( document, [ jQuery ] );
+		whenReady = function( fn ) {
+			readyCallbacks.push( fn );
+
+			if ( !readyFiring ) {
+				readyFiring = true;
+				while ( readyCallbacks.length ) {
+					fn = readyCallbacks.shift();
+					if ( jQuery.isFunction( fn ) ) {
+						fn.call( document, jQuery );
+					}
+				}
+				readyFiring = false;
+			}
+		};
+
+		whenReady();
 	}
 } );
 
+// Make jQuery.ready promise compatible (gh-1778)
```

Comment:
I'd update this to "promise-consumable"; it's more clear and might head off too much effort being spent on bugs like `jQuery.ready.then( fn ).then(…)` not working as expected.